tmt/base.py

```diff
@@ -1191,6 +1200,7 @@ def _filters_conditions(self, nodes, filters, conditions, links):
                 # Handle broken link as not matching
                 continue
             result.append(node)
+        result = sorted(result, key=lambda n: n.order)
```

Comment:
This works fine for most cases, but there's one special case when tests are duplicated:

```yaml
/plan:
    discover:
        how: fmf
        test:
          - test
          - test
    execute:
        how: tmt

/test:
    test: /bin/true

    /one:
        order: 1
    /two:
        order: 2
    /three:
        order: 3
```

This results in the following test discovery:

```
> tmt run discover -v
/var/tmp/tmt/run-056

/plan
    discover
        how: fmf
        directory: /tmp/dup
        tests: test and test
        summary: 6 tests selected
            /test/one
            /test/one
            /test/two
            /test/two
            /test/three
            /test/three
```

We need the following order:

```
/test/one
/test/two
/test/three
/test/one
/test/two
/test/three
```

Reply:
if `order` is involved can we print something for `discover`?

```
> tmt run discover -v
/var/tmp/tmt/run-056

/plan
    discover
        how: fmf
        directory: /tmp/dup
        tests: test and test
        summary: 6 tests selected [ordered]
            /test/one
            /test/one
            /test/two
            /test/two
            /test/three
            /test/three
```

Reply:
> if `order` is involved can we print something for `discover`?

What do you mean by "if order is involved"? Tests are always somehow ordered. As for now alphabetically.